pandas/tests/reshape/concat/test_empty.py

```diff
@@ -249,3 +249,18 @@ def test_empty_dtype_coerce(self):
         result = concat([df1, df2])
         expected = df1.dtypes
         tm.assert_series_equal(result.dtypes, expected)
+
+    def test_concat_empty_dataframe(self):
+        # 39037
+        df1 = DataFrame(columns=["a", "b"])
+        df2 = DataFrame(columns=["b", "c"])
+        result = concat([df1, df2, df1])
+        expected = DataFrame(columns=['a', 'b', 'c'])
+        tm.assert_frame_equal(result, expected)
+
+        df3 = DataFrame(columns=["a", "b"])
+        df4 = DataFrame(columns=["b"])
+        result = concat([df3, df4])
+        expected = DataFrame(columns=['a', 'b'])
+        tm.assert_frame_equal(result, expected)   
```

Comment:
Look at comments in the issue, seems it is also nice to have type of value checked? https://github.com/pandas-dev/pandas/issues/39037#issuecomment-756748197